src/main/java/build/buildfarm/server/services/OperationsService.java

```diff
@@ -38,22 +38,25 @@ public OperationsService(Instance instance) {
   public void listOperations(
       ListOperationsRequest request, StreamObserver<ListOperationsResponse> responseObserver) {
     int pageSize = request.getPageSize();
-    if (pageSize < 0) {
+    if (pageSize < 0 || pageSize > 100) {
       responseObserver.onError(Status.OUT_OF_RANGE.asException());
       return;
     }
+    if (pageSize == 0) {
+      pageSize = 50;
+    }
 
-    ImmutableList.Builder<Operation> operations = new ImmutableList.Builder<>();
-
-    String nextPageToken =
-        instance.listOperations(pageSize, request.getPageToken(), request.getFilter(), operations);
-
-    responseObserver.onNext(
-        ListOperationsResponse.newBuilder()
-            .addAllOperations(operations.build())
-            .setNextPageToken(nextPageToken)
-            .build());
-    responseObserver.onCompleted();
+    // TODO make async
```

Comment:
TODO for now or later?

Reply:
later. This would need async redis interaction on the instance -> backplane, and that's something to be handled when I try lettuce.